File: 001/pseudoinverse.py
import numpy as np
import time

start_time = time.time()

def run():
    y = np.matrix('0; 1; 4; 9')
    data = np.matrix('1 0; 1 1; 1 2; 1 3')

    transpose = data.getT()
    data =  transpose * data
    y = transpose * y
    inverse = data.getI()
    y = inverse * y

for n in range(10):
    run()
    

print("Python:", (time.time() - start_time) / 10)

File: 001/pseudoinverse.jl
starting_time = time_ns()

function run()
    y = [0; 1; 4; 9]
    data = [1 0; 1 1; 1 2; 1 3]

    transposed_matrix = transpose(data)
    data =  transposed_matrix * data
    y = transposed_matrix * y
    inverse = inv(data)
    y = inverse * y

end

for num in 1:10
    run()
end

final_time = time_ns() - starting_time
print("Julia: ", final_time / 10000000000)